internal/report/report.go

```diff
@@ -433,7 +433,17 @@ func PrintAssembly(w io.Writer, rpt *Report, obj plugin.ObjTool, maxFuncs int) e
 	}
 
 	if len(syms) == 0 {
-		return fmt.Errorf("no matches found for regexp: %s", o.Symbol)
+		if address == nil { // User provides a symbol instead of an address
```

Comment:
"instead of address" in the comment reads a bit strange - as if address is what we expected but user entered symbol pattern. Both are OK and there is no primary expected input I think.

Also instead of "provides a symbol" it would be more accurate to say "symbol regexp".

Also "provides" should be "provided".

I maybe would simplify the comment to just: `// The symbol regexp case` and below to `// The address case`.

These are nits, but details are important in good engineering.